### PR TITLE
Handle legacy sql qfile names

### PR DESCRIPTION
**Pull request by szemere**
Handling legacy formatted qfile names in sql driver persist state.

### CODE REVIEW

**Comment by szemere** [comment]
I made some style-check mistakes. Waiting for first reviews before pushing any changes. Sorry.

**Comment by kira-syslogng** [comment]
Build FAILURE

File: lib/persist-state.h [hunk 1]
```diff
@@ -97,6 +97,7 @@ gboolean persist_state_remove_entry(PersistState *self, const gchar *persist_nam
 
 gchar *persist_state_lookup_string(PersistState *self, const gchar *key, gsize *length, guint8 *version);
 gboolean persist_state_rename_entry(PersistState *self, const gchar *old_key, const gchar *new_key);
+void persist_state_copy_entry_value(PersistState *self, const PersistEntryHandle from, PersistEntryHandle to, gsize size);
```
Comment (MrAnno) [inline lib/persist-state.h]: I think we should use `const gchar *old_key, const gchar *new_key` instead, so you could move everything into the persist level and make sure that `size` will be the same for `from` and `to`.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: modules/afsql/afsql.c [hunk 2]
```diff
@@ -980,6 +980,31 @@ afsql_dd_format_persist_name(const LogPipe *s)
...
+_update_legacy_persist_name_if_exists(AFSqlDestDriver *self)
+{
+  GlobalConfig *cfg = log_pipe_get_config(&self->super.super.super.super);
+  const gchar *legacy_persist_name = _afsql_dd_format_legacy_persist_name(self);
+
+  if (!persist_state_entry_exists(cfg->state, legacy_persist_name))
+    return TRUE;
+
+  const gchar *persist_name = afsql_dd_format_persist_name(&self->super.super.super.super);
+  return persist_state_copy_entry(cfg->state, legacy_persist_name, persist_name);
```
Comment (MrAnno) [inline modules/afsql/afsql.c]: move*

Comment (MrAnno) [reply modules/afsql/afsql.c]: This is an undeclared function (implicitly declared) and almost all of our CI jobs are green. Only the CMake job caught this. 😢

File: modules/afsql/afsql.c [hunk 3]
```diff
@@ -980,6 +980,31 @@ afsql_dd_format_persist_name(const LogPipe *s)
   return persist_name;
 }
 
+static const gchar *
+_afsql_dd_format_legacy_persist_name(const AFSqlDestDriver *self)
+{
+  static gchar legacy_persist_name[256];
```
Comment (lbudai) [inline modules/afsql/afsql.c]: Shouldn't we extend this array to a bigger value?

**Comment by lbudai** [comment]
@kira-syslogng : retest this please

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**